src/commonMain/kotlin/com/liftric/job/queue/JobQueue.kt

```diff
@@ -135,18 +145,49 @@ abstract class AbstractJobQueue(
                 job.delegate = delegate
```

Comment:
I guess somewhere in the `while(isActive)` loop we need to invoke `NetworkListener.observeNetworkState()`. The invocation is currently missing in the code :)